### PR TITLE
E2E: add announcement tests

### PR DESCRIPTION
**Pull request by fedepaol**
We add basic announcement tests where all the nodes announce the same
ips to all / some neighbors.

### CODE REVIEW

File: e2etests/tests/advertisement.go [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+				validate: func(frrs []*frrcontainer.FRR, nodes []v1.Node) {
+					ginkgo.By(fmt.Sprintf("checking prefixes on %s", frrs[0].Name))
+					ValidatePrefixesForNeighbor(frrs[0], nodes, "fc00:f853:ccd:e799::/64")
+					ValidateNeighborNoPrefixes(frrs[0], nodes, "fc00:f853:ccd:e800::/64")
+					for _, f := range frrs[1:] {
+						ginkgo.By(fmt.Sprintf("checking prefixes on %s", f.Name))
+						ValidatePrefixesForNeighbor(f, nodes, "fc00:f853:ccd:e799::/64", "fc00:f853:ccd:e800::/64")
+					}
+				},
+			}),
```
Comment (oribon) [inline e2etests/tests/advertisement.go:226]: can we add a dual case at this point or not yet?

File: e2etests/pkg/routes/routes.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+	_, cidr, err := net.ParseCIDR(prefix)
+	if err != nil {
+		return false, err
+	}
+
+	route, err := routeForCIDR(cidr, routesV4, routesV6)
+	var notFound RouteNotFoundError
+	if errors.As(err, &notFound) {
+		return false, nil
+	}
```
Comment (oribon) [inline e2etests/pkg/routes/routes.go:36]: why is this needed?

File: e2etests/pkg/routes/routes.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+	for _, route := range routesV6 {
+		if cidrsAreEqual(route.Destination, cidr) {
+			return route, nil
+		}
+	}
+	return frr.Route{}, RouteNotFoundError(fmt.Sprintf("route %s not found", cidr))
+}
+
+func havePrefix(pod *v1.Pod, pairingFamily ipfamily.Family, prefix *net.IPNet, nextHop, vrf string) bool {
```
Comment (oribon) [inline e2etests/pkg/routes/routes.go:73]: unrelated to this pr?

File: e2etests/tests/advertisement.go [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+				},
+				validate: func(ppV4 []config.Peer, ppV6 []config.Peer, nodes []v1.Node) {
+					ValidatePrefixesForNeighbor(ppV6[0].FRR, nodes, "fc00:f853:ccd:e799::/64")
+					ValidateNeighborNoPrefixes(ppV6[0].FRR, nodes, "fc00:f853:ccd:e800::/64")
+					for _, p := range ppV6[1:] {
+						ValidatePrefixesForNeighbor(p.FRR, nodes, "fc00:f853:ccd:e799::/64", "fc00:f853:ccd:e800::/64")
+					}
+				},
+			}),
+			ginkgo.Entry("DUAL - Advertise with mode allowall", params{
```
Comment (oribon) [inline e2etests/tests/advertisement.go:235]: `DUAL` -> `DUALSTACK` as that is what we skip in the ci? (although this passes on these lanes)

File: e2etests/pkg/routes/routes.go (hunk 2 above)
Comment (fedepaol) [reply e2etests/pkg/routes/routes.go:36]: Because the specific error is different from a conversion error or something not recoverable.
Even though routeForCIDR can only return that type of error, it's better to handle the specific error separately from the generic one, it will make the logic future proof.

File: e2etests/pkg/routes/routes.go (hunk 3 above)
Comment (fedepaol) [reply e2etests/pkg/routes/routes.go:73]: yep, I reworked a lot of the logic in the "incoming prefixes" PR and backported the files, this sneaked in. Removing

**Review by oribon — APPROVED** [review]
lgtm
